### PR TITLE
Support full VM Service URIs

### PR DESCRIPTION
**Pull request by DanTup**

### CODE REVIEW

File: packages/devtools/lib/src/framework/framework_core.dart [hunk 1]
```diff
@@ -65,11 +59,47 @@ class FrameworkCore {
...
+          uri.isAbsolute &&
+          (uri.isScheme('ws') ||
+              uri.isScheme('wss') ||
+              uri.isScheme('http') ||
+              uri.isScheme('https'))) {
+        return uri;
+      }
+    }
+
+    // Otherwise try the legacy port oprtion. Here we assume ws:/localhost and
```
Comment (devoncarew) [inline packages/devtools/lib/src/framework/framework_core.dart]: sp: oprtion

File: packages/devtools/lib/src/model/model.dart [hunk 2]
```diff
@@ -99,9 +99,10 @@ class App {
     return framework.connectDialog.isVisible();
   }
 
-  Future<void> connectDialogConnectTo([dynamic port]) async {
+  Future<void> connectDialogConnectTo([dynamic uri]) async {
+    // uri comes as a String (from JSON) so needs changing back to a URI.
     // ignore: invalid_use_of_visible_for_testing_member
-    return framework.connectDialog.connectTo(port);
+    return framework.connectDialog.connectTo(Uri.parse(uri));
```
Comment (jacob314) [inline packages/devtools/lib/src/model/model.dart]: not your fault but go ahead and make this method not claim to be visible for testing.

Comment (DanTup) [reply packages/devtools/lib/src/model/model.dart]: There are three other instances of this - should I remove them all?

Comment (jacob314) [reply packages/devtools/lib/src/model/model.dart]: Go ahead and remove any ones that aren't in test files.

Comment (DanTup) [reply packages/devtools/lib/src/model/model.dart]: Done!

**Pull request merged**